lib/message_request.js

```diff
@@ -0,0 +1,166 @@
+var logger = require('minilog')('message:request');
+
+var opTable = {
+  control: ['nameSync'],
+  message: ['publish', 'subscribe', 'sync', 'unsubscribe'],
+  presence: ['get', 'set', 'subscribe', 'sync', 'unsubscribe'],
+  status: ['get', 'set', 'subscribe', 'sync', 'unsubscribe'],
+  stream: ['get', 'push', 'subscribe', 'sync']
+};
+
+var Request = function (operation, scope) {
+  this.message = {
+    op: operation,
+    to: scope
+  };
+
+  if (!this._isValid()) {
+    throw new Error('Invalid request message.');
+  }
+};
+
+Request.buildGet = function (scope, options) {
+  return new Request('get', scope).setOptions(options);
+};
+
+Request.buildPublish = function (scope, value) {
+  var request = new Request('publish', scope);
+  request.setAttr('value', value);
+  return request;
+};
+
+Request.buildPush = function (scope, resource, action, value) {
+  var request = new Request('push', scope);
+  request.setAttr('resource', resource);
+  request.setAttr('action', action);
+  request.setAttr('value', value);
+
+  return request;
+};
+
+Request.buildNameSync = function (scope, options) {
+  return new Request('nameSync', scope).setOptions(options);
+};
+
+Request.buildSet = function (scope, value, clientData, key, userType) {
+  var request = new Request('set', scope);
+  request.setAttr('value', value);
+  request.setAttrOptional('key', key);
+  request.setAttrOptional('type', userType);
+  if (typeof(clientData) != 'function') {
+    request.setAttrOptional('clientData', clientData);
+  }
+  return request;
+};
+
+Request.buildSync = function (scope, options) {
+  return new Request('sync', scope).setOptions(options);
+};
+
+Request.buildSubscribe = function (scope, options) {
+  return new Request('subscribe', scope).setOptions(options);
+};
+
+Request.buildUnsubscribe = function (scope, options) {
+  return new Request('unsubscribe', scope);
+};
+
+// TO DO: config class should return the required data via getters
+Request.prototype.setAuthData = function (configuration) {
+  if (configuration && configuration.auth) {
+    this.setAttr('auth', configuration.auth);
+    this.setAttr('userId', configuration.userId);
+    this.setAttr('userType', configuration.userType);
+    this.setAttr('accountName', configuration.accountName);
+  }
+};
+
+Request.getAttr = function (message, attr) {
+  return (message && message[attr]);
+};
+
+// Instance methods
```

Comment:
Instead, moved the method down.  It was once a class method; now an instance method.